istioctl/pkg/util/formatting/formatter.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formatting
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	"istio.io/istio/galley/pkg/config/analysis/diag"
+)
+
+// Formatting options for Messages
+const (
+	LogFormat  = "log"
+	JSONFormat = "json"
+	YAMLFormat = "yaml"
+)
+
+var (
+	MsgOutputFormatKeys = []string{LogFormat, JSONFormat, YAMLFormat}
+	MsgOutputFormats    = make(map[string]bool)
+)
+
+func init() {
+	for _, key := range MsgOutputFormatKeys {
+		MsgOutputFormats[key] = true
+	}
+}
+
+// Print output messages in the specified format with color options
+func Print(ms diag.Messages, format string, colorize bool) (string, error) {
+	switch format {
+	case LogFormat:
+		return printLog(ms, colorize), nil
+	case JSONFormat:
+		return printJSON(ms)
+	case YAMLFormat:
+		return printYAML(ms)
+	default:
+		return "", fmt.Errorf("invalid format, expected one of %v but got %q", MsgOutputFormatKeys, format)
+	}
+}
+
+func printLog(ms diag.Messages, colorize bool) string {
+	var logOutput []string
+	for _, m := range ms {
+		logOutput = append(logOutput, render(m, colorize))
+	}
+	return strings.Join(logOutput, "\n")
+}
+
+func printJSON(ms diag.Messages) (string, error) {
+	jsonOutput, err := json.MarshalIndent(ms, "", "\t")
+	return string(jsonOutput), err
+}
+
+func printYAML(ms diag.Messages) (string, error) {
+	yamlOutput, err := yaml.Marshal(ms)
+	return string(yamlOutput), err
+}
+
+// Formatting options for Message
+var (
+	colorPrefixes = map[diag.Level]string{
+		diag.Info:    "",           // no special color for info messages
+		diag.Warning: "\033[33m",   // yellow
+		diag.Error:   "\033[1;31m", // bold red
+	}
+)
```

Comment:
This looks like it's only used in the colorPrefix function, so it might be worth moving it there. (It's also used in the test file but I addressed that in the file itself.) I don't think making it package-global has any benefits, but I might be wrong.